xblock/fields.py

```diff
@@ -779,6 +779,18 @@ def from_json(self, value):
 
     enforce_type = from_json
 
+    def to_string(self, value):
+        """
+        In python3, json.dumps() cannot sort keys of different types, 
+        so preconvert None to 'null'.
+        """
+        self.enforce_type(value)
+        if isinstance(value, dict) and None in value:
+            value = value.copy()
+            value['null'] = value[None]
+            del value[None]
+        return super(Dict, self).to_string(value)
```

Comment:
I'm not sure if you found http://bugs.python.org/issue25457, but perhaps referencing it when removing `sort_keys`.